drop-analytics/src/moose_impl.rs

```diff
@@ -108,19 +113,33 @@ impl MooseImpl {
         moose_debug!(logger, res, "init");
 
         anyhow::ensure!(res.is_ok(), "Failed to initialize moose: {:?}", res.err());
-
         populate_context(&logger);
-        let res = moose::flush_changes();
-        moose_debug!(logger, res, "flush_changes");
-        res.context("Failed to flush moose context")?;
 
-        Ok(Self { logger })
+        let (tx, rx) = sync_channel(1);
+        let task_logger = logger.clone();
+        let context_update_task = std::thread::spawn(move || loop {
+            match rx.recv_timeout(Duration::from_secs(60 * 10)) {
```

Comment:
Is this correct? I think that on timeout you would receive an error `Timeout` thus this loop will only turn once.
You should probably `populate_context()` on the error and break on `Ok()`.

BTW you don't need to send anything through the channel dropping it suffices, since you can infer it from the error https://doc.rust-lang.org/std/sync/mpsc/enum.TryRecvError.html

Reply:
oh yeah, you're absolutely right on the logic, for the sending part, we'd still need to manually drop the channel to make sure that it happens before joining, so we can't save a code line and explicitly sending the signal looks a little more correct IMO